plugins/kubernetes/app/models/kubernetes/deploy_yaml.rb

```diff
@@ -114,19 +119,45 @@ def set_resource_usage
       }
     end
 
-    def update_docker_image
+    def set_docker_image
       docker_path = @doc.build.docker_repo_digest || "#{@doc.build.project.docker_repo}:#{@doc.build.docker_ref}"
       # Assume first container is one we want to update docker image in
       container.image = docker_path
     end
 
+    def set_env
+      env = (container.env || [])
+
+      # static data
+      metadata = release_doc_metadata
+      [:REVISION, :TAG, :PROJECT, :ROLE, :DEPLOY_ID, :DEPLOY_GROUP].each do |k|
+        env << {name: k, value: metadata.fetch(k.downcase)}
+      end
+
+      # dynamic lookups for unknown things during deploy
+      {
+        POD_NAME: 'metadata.name',
+        POD_NAMESPACE: 'metadata.namespace',
+        POD_IP: 'status.podIP'
+      }.each do |k,v|
+         env << {
+          name: k,
+          valueFrom: {fieldRef: {fieldPath: v}}
+        }
+      end
+
+      container.env = env
```

Comment:
Hmm, I wonder if we should apply these env vars to all the containers in a pod, or just the first one.  Will give that some thought as we start to use sidecars more.